pkg/controllers/observability-ui/observability-ui-plugin/compatibility_matrix.go

```diff
@@ -0,0 +1,52 @@
+package observability_ui_plugin
+
+import (
+	"strings"
+
+	"golang.org/x/mod/semver"
+
+	obsui "github.com/rhobs/observability-operator/pkg/apis/observability-ui/v1alpha1"
+)
+
+type CompatibilityEntry struct {
+	PluginType        obsui.UIPluginType
+	MinClusterVersion string
+	MaxClusterVersion string
+	ImageKey          string
+	Features          []string
+}
+
+var compatibilityMatrix = []CompatibilityEntry{
+	{
+		PluginType:        obsui.TypeDashboards,
+		MinClusterVersion: "v4.11",
+		MaxClusterVersion: "",
+		ImageKey:          "ui-dashboards",
+		Features:          []string{},
+	},
+}
+
+func getImageKeyForPluginType(pluginType obsui.UIPluginType, clusterVersion string) string {
```

Comment:
(nit) the function could return (string, error) with error != nil if no key is found.